### PR TITLE
MEC028 (wais) v2.1.10 update and StaDataRate subscription

### PR DESCRIPTION
**Pull request by pastorsx**
MEC028 update with 2.1.10 data structures and endpoints
- compile previous code and report notification as it used to
- need a follow-up update to have the notification matching the new subscription triggers and threshold (subscription behavior changed'
- added staDataRate subscription
  - SBI updates to track apps under a UE
  - update to the code to get the downlink and uplink traffic from the metric store
  - storage of the downlink and uplink metrics
  - notification based on updates to the scenario (same as assocStaNotification). Valid but missing the periodic trigger (to be done as part of the assocSta update - same periodic trigger)

testing
- scenario using wifi POAs, notification generated as expected based on triggers
- ran the current system test (passed).

Code update
- lots of file changed caused by the new yaml file being generated
- real changes limited to :
    go-apps/meep-wais/sbi/wais-sbi.go
    go-apps/meep-wais/server/wais.go
    go-apps/meep-wais/server/wais_test.go
    go-apps/meep-wais/server/convert.go
    go-apps/meep-wais/api/swagger.yaml
    test/system/wais_test.go

### CODE REVIEW

File: go-apps/meep-wais/server/wais.go [hunk 1]
```diff
@@ -178,49 +198,76 @@ func Stop() (err error) {
...
+	if jsonStaData != "" {
+		staData = convertJsonToStaData(jsonStaData)
+	}
+
+	//update data rates
+	sumUl := 0.0
+	sumDl := 0.0
+	for _, appName := range appNames {
+
+		metricsArray, err := metricStore.GetCachedNetworkMetrics("*", appName)
```
Comment (dilallkx) [inline go-apps/meep-wais/server/wais.go:215]: This should be done in the SBI.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-apps/meep-wais/server/wais.go (hunk 1 above)
Comment (pastorsx) [reply go-apps/meep-wais/server/wais.go:215]: this is the function updateStaInfo which is called from the sbi but implemented in the wais.go file.... So i think that is the SBI. can you confirm you agree?

Comment (dilallkx) [reply go-apps/meep-wais/server/wais.go:215]: Yes, the SBI invokes this function, however the wais now depends on the metric store to retrieve the network metrics. I believe this dependency should be in the SBI package only. The SBI package should be the only one that changes to obtain the necessary data.

Comment (pastorsx) [reply go-apps/meep-wais/server/wais.go]: moved the metricStore code to the sbi

File: go-apps/meep-wais/sbi/wais-sbi.go [hunk 2]
```diff
@@ -195,8 +223,17 @@ func processActiveScenarioUpdate() {
...
+	sbi.updateScenarioNameCB(scenarioName, &update)
+
+	// Connect to Metric Store
+	if update {
+		var err error
+		metricStore, err = met.NewMetricStore(scenarioName, sbi.sandboxName, influxAddr, redisAddr)
+		if err != nil {
+			log.Error("Failed connection to metric-store: ", err)
+		}
+	}
```
Comment (dilallkx) [inline go-apps/meep-wais/sbi/wais-sbi.go:236]: The SBI should not depend on the WAIS to determine when to connect to the metric store.

**Review by dilallkx — CHANGES_REQUESTED** [review]

**Comment by dilallkx** [comment]
Also, please rebase this branch to the tip of develop.

File: go-apps/meep-wais/sbi/wais-sbi.go (hunk 2 above)
Comment (pastorsx) [reply go-apps/meep-wais/sbi/wais-sbi.go]: i put all the logic in sbi now

**Review by dilallkx — APPROVED** [review]

**Pull request merged**